src/platform/stm32/libs/libusbhost/src/usbh_core.c

```diff
@@ -415,35 +416,31 @@ static void device_enumerate(usbh_device_t *dev, usbh_packet_callback_data_t cb_
 		}
 		break;
 
-	case USBH_ENUM_STATE_DEVICE_DT_READ_SETUP:
+	case USBH_ENUM_STATE_DEVICE_DT_READ_SHORT_SETUP:
 		{
 			struct usb_setup_data setup_data;
 
 			setup_data.bmRequestType = USB_REQ_TYPE_IN | USB_REQ_TYPE_DEVICE;
 			setup_data.bRequest = USB_REQ_GET_DESCRIPTOR;
 			setup_data.wValue = USB_DT_DEVICE << 8;
 			setup_data.wIndex = 0;
-			setup_data.wLength = USB_DT_DEVICE_SIZE;
-
-			dev->state = USBH_ENUM_STATE_DEVICE_DT_READ_COMPLETE;
+			setup_data.wLength = 8;
+			dev->state = USBH_ENUM_STATE_DEVICE_DT_READ_SHORT_COMPLETE;
 			device_control(dev, device_enumerate, &setup_data, &usbh_buffer[0]);
 		}
 		break;
 
-	case USBH_ENUM_STATE_DEVICE_DT_READ_COMPLETE:
+	case USBH_ENUM_STATE_DEVICE_DT_READ_SHORT_COMPLETE:
 		{
 			switch (cb_data.status) {
 			case USBH_PACKET_CALLBACK_STATUS_OK:
 				{
 					struct usb_device_descriptor *ddt =
 							(struct usb_device_descriptor *)&usbh_buffer[0];
 					dev->packet_size_max0 = ddt->bMaxPacketSize0;
-					LOG_PRINTF("Found device with vid=0x%04x pid=0x%04x\n", ddt->idVendor, ddt->idProduct);
-					LOG_PRINTF("class=0x%02x subclass=0x%02x protocol=0x%02x\n", ddt->bDeviceClass, ddt->bDeviceSubClass, ddt->bDeviceProtocol);
-					CONTINUE_WITH(USBH_ENUM_STATE_CONFIGURATION_DT_HEADER_READ_SETUP)
+					CONTINUE_WITH(USBH_ENUM_STATE_DEVICE_DT_READ_SETUP);
 				}
 				break;
-
 			case USBH_PACKET_CALLBACK_STATUS_ERRSIZ:
```

Comment:
Now that we only send 8 bytes (assuming that's the required minimum by the standard?!?) can we ever hit this? I guess it's still save to continue just wondering.

Reply:
I'm not *that* deep into USB and this lib, and I don't know what weird USB devices are out there, so I guess if we get 8 bytes we should be happy and move on.

Reply:
USBH_PACKET_CALLBACK_STATUS_ERRSIZ gets set if the returned data is not *exactly* the right size, I guess that would trigger if a device returns 18 when asked for 8, so better to leave it in

Reply:
I agree.